### PR TITLE
OAuth registration in GUI

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: gui/app/main.js [hunk 1]
```diff
@@ -70,6 +69,7 @@ const sendToMainWindow = (...args) => {
 }
 
 const sendErrorToMainWindow = (msg) => {
+  // FIXME
```
Comment (sebn) [inline gui/app/main.js]: Why?

File: gui/app/main.js [hunk 2]
```diff
@@ -464,17 +446,16 @@ ipcMain.on('choose-folder', (event) => {
 })
 
 ipcMain.on('start-sync', (event, arg) => {
-  if (!device) {
+  if (!desktop.config.client) {
     console.error('No device!')
```
Comment (sebn) [inline gui/app/main.js]: `'No client!'` ?

Comment (sebn) [inline gui/app/main.js]: Would `if (!desktop.config.isValid())` make sense?

File: gui/app/main.js [hunk 3]
```diff
@@ -414,25 +423,24 @@ app.on('ready', () => {
...
-})
-
 ipcMain.on('register-remote', (event, arg) => {
-  // TODO use package.json for pkg ({} below)
-  desktop.config.cozyUrl = arg.url
-  desktop.registerRemote(arg.url, {})
+  desktop.config.cozyUrl = arg.cozyUrl
+  const onRegistered = (client, url) => {
+    let resolveP
+    const promise = new Promise((resolve) => { resolveP = resolve })
```
Comment (sebn) [inline gui/app/main.js]: Oh my :)
I can't think of a better solution yet.

File: gui/app/main.js (hunk 1 above)
Comment (nono) [inline gui/app/main.js]: The error messages won't be the same with cozy-client-js

File: gui/app/main.js [hunk 4]
```diff
@@ -454,8 +454,8 @@ ipcMain.on('choose-folder', (event) => {
 })
 
 ipcMain.on('start-sync', (event, arg) => {
-  if (!desktop.config.client) {
-    console.error('No device!')
+  if (!desktop.config.isValid()) {
+    console.error('No client!')
```
Comment (sebn) [inline gui/app/main.js]: `'No valid config'` then (sorry)

**Comment by sebn** [comment]
👍

**Pull request merged**